src/shaders/symbol_sdf.vertex.glsl

```diff
@@ -43,8 +58,35 @@ void main() {
     mediump float a_minzoom = a_zoom[0];
     mediump float a_maxzoom = a_zoom[1];
 
-    // u_zoom is the current zoom level adjusted for the change in font size
-    mediump float z = 2.0 - step(a_minzoom, u_zoom) - (1.0 - step(a_maxzoom, u_zoom));
+    // In order to accommodate placing labels around corners in
+    // symbol-placement: line, each glyph in a label could have multiple layout
+    // "instances", only one of which should be shown at a given zoom level.
+    // The min/max zoom assigned to each instance is based on the font size at
+    // the vector tile's zoom level, which might be different than at the
+    // currently rendered zoom level if text-size is zoom-dependent.
+    // Thus, we compensate for this difference by calculating an adjustment
+    // based on the scale of rendered text size relative to layout text size.
+    mediump float adjustedSize;
+    if (!u_is_size_zoom_constant && !u_is_size_feature_constant) {
+        v_size = mix(a_size[0], a_size[1], u_size_t) / 10.0;
+        adjustedSize = a_size[2] / 10.0;
+    } else if (u_is_size_zoom_constant && !u_is_size_feature_constant) {
+        v_size = a_size[0] / 10.0;
+        adjustedSize = v_size;
+    } else if (!u_is_size_zoom_constant && u_is_size_feature_constant) {
+        v_size = u_size;
+        adjustedSize = u_adjusted_size;
+    } else {
+        v_size = u_size;
+        adjustedSize = u_size;
+    }
+
+    float fontScale = u_is_text ? v_size / 24.0 : v_size;
+
+    mediump float zoomAdjust = log2(v_size / adjustedSize);
+    mediump float adjustedZoom = (u_zoom - zoomAdjust) * 10.0;
+    // result: z = 0 if a_minzoom <= adjustedZoom < a_maxzoom, and 1 otherwise
```

Comment:
Yeah, I _think_ that's right.  I wasn't 100% sure of the clip space part, though, and so didn't include it in the comment in case that was incorrect and thus misleading.  But if that's your understanding, too, I think that's confirmation enough...